core/templates/pages/collection-editor-page/services/collection-editor-state.service.spec.ts

```diff
@@ -17,203 +17,107 @@
-    mockCollectionRightsBackendApiService =
-    new MockCollectionRightsBackendApiService();
-    collectionUpdateService = TestBed.inject(CollectionUpdateService);
-
-    mockEditableCollectionBackendApiService.newBackendCollectionObject = {
-      id: '0',
-      title: 'Collection Under Test',
-      category: 'Test',
-      objective: 'To pass',
-      language_code: 'en',
-      schema_version: '3',
-      version: '1',
-      nodes: [{
-        exploration_id: '0'
-      }, {
-        exploration_id: '1'
-      }],
+    undoRedoService = TestBed.inject(UndoRedoService);
+    editableCollectionBackendApiService = TestBed.inject(
+      EditableCollectionBackendApiService);
+    collectionRightsBackendApiService = TestBed.inject(
+      CollectionRightsBackendApiService);
+    alertsService = TestBed.inject(AlertsService);
+
+    sampleCollectionBackendDict = {
+      id: 'sample_collection_id',
+      title: 'a title',
+      objective: 'an objective',
+      category: 'a category',
+      version: 1,
+      nodes: [],
+      language_code: null,
+      schema_version: null,
+      tags: null,
       playthrough_dict: {
         next_exploration_id: 'expId',
         completed_exploration_ids: ['expId2']
       }
     };
 
-    secondBackendCollectionObject = {
-      id: '5',
-      title: 'Interesting collection',
-      category: 'Test',
-      objective: 'To be interesting',
-      language_code: 'en',
-      tags: [],
-      schema_version: '3',
-      version: '3',
-      nodes: [{
-        exploration_id: '0'
-      }],
-      playthrough_dict: {
-        next_exploration_id: 'expId',
-        completed_exploration_ids: ['expId2']
-      }
-    };
+    sampleCollection = Collection.create(sampleCollectionBackendDict);
 
-    let privateCollectionRightsObject = {
-      collection_id: '5',
-      can_edit: 'true',
-      can_unpublish: 'false',
-      is_private: 'true',
+    sampleCollectionRightsDict = {
+      collection_id: 0,
+      can_edit: true,
+      can_unpublish: false,
+      is_private: true,
       owner_names: ['A']
     };
 
-    mockCollectionRightsBackendApiService.backendCollectionRightsObject =
-privateCollectionRightsObject;
-
-    unpublishablePublicCollectionRightsObject = {
-      collection_id: '5',
-      can_edit: 'true',
-      can_unpublish: 'true',
-      is_private: 'false',
-      owner_names: ['A']
-    };
+    sampleCollectionRights = CollectionRights.create(
+      sampleCollectionRightsDict);
   }));
 
   beforeEach(() => {
     testSubscriptions = new Subscription();
     testSubscriptions.add(
       collectionEditorStateService.onCollectionInitialized.subscribe(
         collectionInitializedSpy));
+
+    alertsSpy = spyOn(alertsService, 'addWarning').and.returnValue(null);
   });
 
   afterEach(() => {
     testSubscriptions.unsubscribe();
   });
 
-  it('should request to load the collection from the backend', () => {
```

Comment:
Why were these tests deleted?

Reply:
Previously this spec is not tested correctly, so I had to revamp the entire file.
(Some functions are not even being called.)